### PR TITLE
Add logging to cross-browser example

### PR DESCRIPTION
**Pull request by mathiasbynens**
Issue: #110

### CODE REVIEW

File: examples/cross-browser.py [hunk 1]
```diff
@@ -16,13 +16,19 @@
...
 import asyncio
 import json
+import logging
 import os
 import websockets
-from pathlib import Path
 
 
+logging.basicConfig(
+    format="%(message)s",
```
Comment (sadym-chromium) [inline examples/cross-browser.py:28]: is the format needed?

**Review by sadym-chromium — APPROVED** [review]
It's fine as soon as it works. Not sure if the custom `format` is needed though.

File: examples/cross-browser.py (hunk 1 above)
Comment (mathiasbynens) [reply examples/cross-browser.py:28]: If we don’t specify a format it uses the default, which looks like this:

```
$ python3 examples/cross-browser.py
DEBUG:asyncio:Using selector: KqueueSelector
DEBUG:websockets.protocol:client - state = CONNECTING
DEBUG:websockets.protocol:client - event = connection_made(<_SelectorSocketTransport fd=8 read=idle write=<idle, bufsize=0>>)
DEBUG:websockets.client:client > GET /session HTTP/1.1
DEBUG:websockets.client:client > Headers([('Host', 'localhost:8080'), ('Upgrade', 'websocket'), ('Connection', 'Upgrade'), ('Sec-WebSocket-Key', 'f4uln8Ur74ROEp2fJhQD3A=='), ('Sec-WebSocket-Version', '13'), ('Sec-WebSocket-Extensions', 'permessage-deflate; client_max_window_bits'), ('User-Agent', 'Python/3.9 websockets/8.1')])
DEBUG:websockets.protocol:client - event = data_received(<129 bytes>)
DEBUG:websockets.client:client < HTTP/1.1 101 Switching Protocols
DEBUG:websockets.client:client < Headers([('Upgrade', 'websocket'), ('Connection', 'Upgrade'), ('Sec-WebSocket-Accept', 'xBnd9Iuniv0/8O75ePDZZfGAPBE=')])
DEBUG:websockets.protocol:client - state = OPEN
DEBUG:websockets.protocol:client > Frame(fin=True, opcode=1, data=b'{"id": 1000, "method": "browsingContext.create", "params": {}}', rsv1=False, rsv2=False, rsv3=False)
DEBUG:websockets.protocol:client - event = data_received(<92 bytes>)
DEBUG:websockets.protocol:client < Frame(fin=True, opcode=1, data=b'{"id":1000,"result":{"context":"CBEC4DEEA4B2EC3A864A361E16997651","url":"","children":[]}}', rsv1=False, rsv2=False, rsv3=False)
DEBUG:websockets.protocol:client > Frame(fin=True, opcode=1, data=b'{"id": 1001, "method": "browsingContext.navigate", "params": {"url": "file:///Users/mathiasb/projects/chromium-bidi/examples/app.html", "context": "CBEC4DEEA4B2EC3A864A361E16997651"}}', rsv1=False, rsv2=False, rsv3=False)
DEBUG:websockets.protocol:client - event = data_received(<146 bytes>)
DEBUG:websockets.protocol:client < Frame(fin=True, opcode=1, data=b'{"id":1001,"result":{"navigation":"73D539DF67D63824ADAC5CE9DEFD3A87","url":"file:///Users/mathiasb/projects/chromium-bidi/examples/app.html"}}', rsv1=False, rsv2=False, rsv3=False)
DEBUG:websockets.protocol:client > Frame(fin=True, opcode=1, data=b'{"id": 1002, "method": "script.callFunction", "params": {"functionDeclaration": "(resultsSelector) => {\\n                const anchors = Array.from(document.querySelectorAll(resultsSelector));\\n                return anchors.map((anchor) => {\\n                    const title = anchor.textContent.trim();\\n                    return `${title} - ${anchor.href}`;\\n                });\\n            }", "args": [{"type": "string", "value": ".titlelink"}], "target": {"context": "CBEC4DEEA4B2EC3A864A361E16997651"}}}', rsv1=False, rsv2=False, rsv3=False)
DEBUG:websockets.protocol:client - event = data_received(<867 bytes>)
```

With the format in the original patch, it looks like this:

```
$ python3 examples/cross-browser.py
Using selector: KqueueSelector
client - state = CONNECTING
client - event = connection_made(<_SelectorSocketTransport fd=8 read=idle write=<idle, bufsize=0>>)
client > GET /session HTTP/1.1
client > Headers([('Host', 'localhost:8080'), ('Upgrade', 'websocket'), ('Connection', 'Upgrade'), ('Sec-WebSocket-Key', '2N+ARhFE+D7KCSPusJK2nQ=='), ('Sec-WebSocket-Version', '13'), ('Sec-WebSocket-Extensions', 'permessage-deflate; client_max_window_bits'), ('User-Agent', 'Python/3.9 websockets/8.1')])
client - event = data_received(<129 bytes>)
client < HTTP/1.1 101 Switching Protocols
client < Headers([('Upgrade', 'websocket'), ('Connection', 'Upgrade'), ('Sec-WebSocket-Accept', '1wDZSv9AfQmTLuWLFM2/h/+hSeU=')])
client - state = OPEN
client > Frame(fin=True, opcode=1, data=b'{"id": 1000, "method": "browsingContext.create", "params": {}}', rsv1=False, rsv2=False, rsv3=False)
client - event = data_received(<92 bytes>)
client < Frame(fin=True, opcode=1, data=b'{"id":1000,"result":{"context":"DE7A751E4DE469A65C5941C1534FC52D","url":"","children":[]}}', rsv1=False, rsv2=False, rsv3=False)
client > Frame(fin=True, opcode=1, data=b'{"id": 1001, "method": "browsingContext.navigate", "params": {"url": "file:///Users/mathiasb/projects/chromium-bidi/examples/app.html", "context": "DE7A751E4DE469A65C5941C1534FC52D"}}', rsv1=False, rsv2=False, rsv3=False)
client - event = data_received(<146 bytes>)
client < Frame(fin=True, opcode=1, data=b'{"id":1001,"result":{"navigation":"7C866424FCFE5593B327D82F8F4D7ABE","url":"file:///Users/mathiasb/projects/chromium-bidi/examples/app.html"}}', rsv1=False, rsv2=False, rsv3=False)
client > Frame(fin=True, opcode=1, data=b'{"id": 1002, "method": "script.callFunction", "params": {"functionDeclaration": "(resultsSelector) => {\\n                const anchors = Array.from(document.querySelectorAll(resultsSelector));\\n                return anchors.map((anchor) => {\\n                    const title = anchor.textContent.trim();\\n                    return `${title} - ${anchor.href}`;\\n                });\\n            }", "args": [{"type": "string", "value": ".titlelink"}], "target": {"context": "DE7A751E4DE469A65C5941C1534FC52D"}}}', rsv1=False, rsv2=False, rsv3=False)
client - event = data_received(<8225 bytes>)
client < Frame(fin=True, opcode=1, data=b'{"id":1002,"result":{"objectId":"29970022457830403.47782690302618236.053992775839285745","type":"array","value":[{"type":"string","value":"No, you can\xe2\x80\x99t save \xc2\xa330 per year by switching off your \xe2\x80\x9cstandby\xe2\x80\x9d devices - https://shkspr.mobi/blog/2021/10/no-you-cant-save-30-per-year-by-switching-off-your-standby-devices/"},{"type":"string","value":"MIT researchers develop a paper thin loudspeaker - https://news.mit.edu/2022/low-power-thin-loudspeaker-0426"},{"type":"string","value":"Khan Academy launches Khan World School online high school - https://asuprep.asu.edu/khan-world-school/"},{"type":"string","value":"SELinux is unmanageable; just turn it off if it gets in your way - https://www.ctrl.blog/entry/selinux-unmanageable.html"},{"type":"string","value":"Statistical Analysis shows Echos process voice to serve ads - https://arxiv.org/abs/2204.10920"},{"type":"string","value":"FFmpeg now supports JPEG XL - https://git.videolan.org/?p=ffmpeg.git;a=commit;h=0008c159562b877700cd9b7c96f941de4ee69af5"},{"type":"string","value":"Playhouse (YC S21) Is Hiring a Founding Full-Stack Engineer - https://www.ycombinator.com/companies/playhouse/jobs/s9TaYfC-founding-full-stack-engineer"},{"type":"string","value":"Tutorial Series to learn Common Lisp quickly - https://github.com/rabbibotton/clog/blob/main/LEARN.md"},{"type":"string","value":"Introduction to Haskell Typeclasses - https://serokell.io/blog/haskell-typeclasses"},{"type":"string","value":"Textsynth: Bellard\'s free GPT-NeoX-20B, GPT-J playground and paid API - https://textsynth.com/playground.html"},{"type":"string","value":"What Lurks in a Drowned Forest in Alabama? - https://nautil.us/what-lurks-in-a-drowned-forest-in-alabama-16508/"},{"type":"string","value":"Four Eras of JavaScript Frameworks - https://www.pzuraq.com/blog/four-eras-of-javascript-frameworks"},{"type":"string","value":"Reversing a 2.4GHz Remote Control - https://xakcop.com/post/re-2.4ghz/"},{"type":"string","value":"The Datasette Ecosystem - https://docs.datasette.io/en/stable/ecosystem.html"},{"type":"string","value":"Uber concedes deception, prepares for $26m ACCC spanking - https://www.innovationaus.com/uber-concedes-deception-prepares-for-26m-accc-spanking/"},{"type":"string","value":"Datalog in JavaScript - https://www.instantdb.dev/essays/datalogjs"},{"type":"string","value":"Show HN: Lists.sh \xe2\x80\x93 A Microblog for Lists - https://lists.sh/"},{"type":"string","value":"Stalin\xe2\x80\x99s Architect by Deyan Sudjic review \xe2\x80\x93 a monumental life - https://www.theguardian.com/books/2022/apr/18/stalins-architect-by-deyan-sudjic-review-a-momumental-life"},{"type":"string","value":"Google Scanned Objects: A High-Quality Dataset of 3D Scanned Household Items - https://arxiv.org/abs/2204.11918"},{"type":"string","value":"How to write a Git commit message (2014) - https://cbea.ms/git-commit/"},{"type":"string","value":"A Brief History of High Availability - https://www.cockroachlabs.com/blog/brief-history-high-availability/"},{"type":"string","value":"Esports stars have shorter careers than NFL players - https://www.washingtonpost.com/video-games/esports/2022/04/19/esports-age-retirement/"},{"type":"string","value":"Show HN: Baseten \xe2\x80\x93 Build ML-powered applications - https://www.baseten.co/"},{"type":"string","value":"What\xe2\x80\x99s Behind the Naming of TBD54566975? - https://blog.concannon.tech/tech-talk/tbd54566975/"},{"type":"string","value":"Payment startup Bolt sued by its most prominent customer - https://www.bloomberg.com/news/articles/2022-04-26/forever-21-parent-authentic-brands-sues-bolt-startup"},{"type":"string","value":"Infinite Mac - https://macos8.app/"},{"type":"string","value":"HTML event handler attributes: down the rabbit hole - https://datastation.multiprocess.io/blog/2022-04-26-event-handler-attributes.html"},{"type":"string","value":"Vectorization in OLAP Databases - https://aneesh.mataroa.blog/blog/vectorization-in-olap-databases/"},{"type":"string","value":"Now I\xe2\x80\x99m calculating with constructive reals - https://aperiodical.com/2022/03/now-im-calculating-with-constructive-reals/"},{"type":"string","value":"The Mysteries of Lisp (2015) - https://arxiv.org/abs/1505.07375"},{"type":"string","value":"No, you can\xe2\x80\x99t save \xc2\xa330 per year by switching off your \xe2\x80\x9cstandby\xe2\x80\x9d devices - https://shkspr.mobi/blog/2021/10/no-you-cant-save-30-per-year-by-switching-off-your-standby-devices/"},{"type":"string","value":"MIT researchers develop a paper thin loudspeaker - https://news.mit.edu/2022/low-power-thin-loudspeaker-0426"},{"type":"string","value":"Khan Academy launches Khan World School online high school - https://asuprep.asu.edu/khan-world-school/"},{"type":"string","value":"SELinux is unmanageable; just turn it off if it gets in your way - https://www.ctrl.blog/entry/selinux-unmanageable.html"},{"type":"string","value":"Statistical Analysis shows Echos process voice to serve ads - https://arxiv.org/abs/2204.10920"},{"type":"string","value":"FFmpeg now supports JPEG XL - https://git.videolan.org/?p=ffmpeg.git;a=commit;h=0008c159562b877700cd9b7c96f941de4ee69af5"},{"type":"string","value":"Playhouse (YC S21) Is Hiring a Founding Full-Stack Engineer - https://www.ycombinator.com/companies/playhouse/jobs/s9TaYfC-founding-full-stack-engineer"},{"type":"string","value":"Tutorial Series to learn Common Lisp quickly - https://github.com/rabbibotton/clog/blob/main/LEARN.md"},{"type":"string","value":"Introduction to Haskell Typeclasses - https://serokell.io/blog/haskell-typeclasses"},{"type":"string","value":"Textsynth: Bellard\'s free GPT-NeoX-20B, GPT-J playground and paid API - https://textsynth.com/playground.html"},{"type":"string","value":"What Lurks in a Drowned Forest in Alabama? - https://nautil.us/what-lurks-in-a-drowned-forest-in-alabama-16508/"},{"type":"string","value":"Four Eras of JavaScript Frameworks - https://www.pzuraq.com/blog/four-eras-of-javascript-frameworks"},{"type":"string","value":"Reversing a 2.4GHz Remote Control - https://xakcop.com/post/re-2.4ghz/"},{"type":"string","value":"The Datasette Ecosystem - https://docs.datasette.io/en/stable/ecosystem.html"},{"type":"string","value":"Uber concedes deception, prepares for $26m ACCC spanking - https://www.innovationaus.com/uber-concedes-deception-prepares-for-26m-accc-spanking/"},{"type":"string","value":"Datalog in JavaScript - https://www.instantdb.dev/essays/datalogjs"},{"type":"string","value":"Show HN: Lists.sh \xe2\x80\x93 A Microblog for Lists - https://lists.sh/"},{"type":"string","value":"Stalin\xe2\x80\x99s Architect by Deyan Sudjic review \xe2\x80\x93 a monumental life - https://www.theguardian.com/books/2022/apr/18/stalins-architect-by-deyan-sudjic-review-a-momumental-life"},{"type":"string","value":"Google Scanned Objects: A High-Quality Dataset of 3D Scanned Household Items - https://arxiv.org/abs/2204.11918"},{"type":"string","value":"How to write a Git commit message (2014) - https://cbea.ms/git-commit/"},{"type":"string","value":"A Brief History of High Availability - https://www.cockroachlabs.com/blog/brief-history-high-availability/"},{"type":"string","value":"Esports stars have shorter careers than NFL players - https://www.washingtonpost.com/video-games/esports/2022/04/19/esports-age-retirement/"},{"type":"string","value":"Show HN: Baseten \xe2\x80\x93 Build ML-powered applications - https://www.baseten.co/"},{"type":"string","value":"What\xe2\x80\x99s Behind the Naming of TBD54566975? - https://blog.concannon.tech/tech-talk/tbd54566975/"},{"type":"string","value":"Payment startup Bolt sued by its most prominent customer - https://www.bloomberg.com/news/articles/2022-04-26/forever-21-parent-authentic-brands-sues-bolt-startup"},{"type":"string","value":"Infinite Mac - https://macos8.app/"},{"type":"string","value":"HTML event handler attributes: down the rabbit hole - https://datastation.multiprocess.io/blog/2022-04-26-event-handler-attributes.html"},{"type":"string","value":"Vectorization in OLAP Databases - https://aneesh.mataroa.blog/blog/vectorization-in-olap-databases/"},{"type":"string","value":"Now I\xe2\x80\x99m calculating with constructive reals - https://aperiodical.com/2022/03/now-im-calculating-with-constructive-reals/"},{"type":"string","value":"The Mysteries of Lisp (2015) - https://arxiv.org/abs/1505.07375"}]}}', rsv1=False, rsv2=False, rsv3=False)
```

Happy to use the default.

Comment (sadym-chromium) [reply examples/cross-browser.py]: Agree, the custom one looks better )

**Pull request merged**